graphql_compiler/global_utils.py

```diff
@@ -59,3 +59,25 @@ def is_same_type(left, right):
         return is_same_type(left.of_type, right.of_type)
     else:
         return False
+
+
+_KeyType = TypeVar("_KeyType")
+
+
+def validate_that_dicts_have_the_same_keys(
+    dict1: Dict[_KeyType, Any], dict2: Dict[_KeyType, Any]
```

Comment:
I don't think this asserts the types are the same. It only asserts that if the key type in dict1 is one of the allowed types (any type), and the key type in dict2 is one of the allowed types (any type), then the returned value is None. So I think {1: 5} and {True: 8} would be considered to have the same keys. Might as well make the key `str` until there's a use case for a generic function.